### PR TITLE
IOBUF inference using Yosys

### PR DESCRIPTION
**Pull request by mkurc-ant**
Fixes https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1161

### CODE REVIEW

File: xc7/yosys/synth.tcl [hunk 1]
```diff
@@ -1,5 +1,21 @@
 yosys -import
 
+# Infer 3-state IOBUFs.
```
Comment (litghost) [inline xc7/yosys/synth.tcl]: I don't have a problem with any of this, I'm wondering if this should be moved into `synth_xilinx` in upstream yosys?  The changes all seem to be independent of symbiflow-arch-defs?

Comment (mkurc-ant) [reply xc7/yosys/synth.tcl]: You're probably right. I'll check what steps does the `synth_xilinx` command do regarding IOs and see how I can augment it.

In my opinion the fact that i have to map `$tribuf` to `$_TBUF_` "manually" prior to the `iopadmap` is kind of a workaround. Besides that the `iopadmap` is not behaving properly as it always leaves some dangling wires connected directly to the io pad.

Comment (litghost) [reply xc7/yosys/synth.tcl]: Part of the reason I think this change might belong to upstream yosys is they can comment whether what you are doing is the "proper" way.  I am not really equiped to say one way or another.

Comment (mithro) [reply xc7/yosys/synth.tcl]: @mkurc-ant I think the correct person to talk to is @mwkmwkmwk who has been doing a lot of work in this area?

Comment (mkurc-ant) [reply xc7/yosys/synth.tcl]: @mithro Ok, I'll do. Meanwhile the only example of how to use the `iopadmap` I found is that reddit post: https://www.reddit.com/r/yosys/comments/4hwlkx/new_support_for_tristate_buffers_in_iopadmap/

**Comment by mkurc-ant** [comment]
There are two recent PRs to the upstream Yosys that addresses the issue I'm having with the inference (https://github.com/YosysHQ/yosys/pull/1527, https://github.com/YosysHQ/yosys/pull/1528). I've modified the Yosys script to use the `-iopad` option of `synth_xilinx` and when used with yosys that has those two PRs merged it works.

I've created a PR to our Yosys fork (https://github.com/SymbiFlow/yosys/pull/45) which contains those changes.

We can either wait for those PRs to get merged upstream and use the upstream Yosys or merge https://github.com/SymbiFlow/yosys/pull/45 and use out master+wip one.

**Comment by litghost** [comment]
With https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1195, I believe you can rebase this PR and the inferred IOBUF should work?

File: xc7/yosys/synth.tcl [hunk 2]
```diff
@@ -2,9 +2,7 @@ yosys -import
 
 # -flatten is used to ensure that the output eblif has only one module.
 # Some of symbiflow expects eblifs with only one module.
-synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp
-
-write_verilog $::env(OUT_SYNTH_V).premap.v
+synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad
```
Comment (litghost) [inline xc7/yosys/synth.tcl]: Why did you remove the premap output?

Comment (mkurc-ant) [reply xc7/yosys/synth.tcl]: I must have done that by accident. Fixed.
